cgl.h

```diff
@@ -144,34 +144,59 @@ CGL_net_addrinfo* CGL_net_addrinfo_query(const char* name, const char* port, siz
 void CGL_net_addrinfo_destroy(CGL_net_addrinfo* infos);
 
 CGL_net_socket* CGL_net_socket_create();
-bool CGL_net_socket_connect(CGL_net_socket* socket, CGL_net_addrinfo* target);
-bool CGL_net_socket_bind(CGL_net_socket* socket, CGL_net_addrinfo* target);
-bool CGL_net_socket_listen(CGL_net_socket* socket, size_t max_connections); 
-CGL_net_socket* CGL_net_socket_accept(CGL_net_socket* socket, CGL_net_addrinfo* addrinfo);
-void CGL_net_socket_close(CGL_net_socket* socket);
-bool CGL_net_socket_send(CGL_net_socket* socket, void* buffer, size_t size, size_t* size_sent);
-bool CGL_net_socket_recv(CGL_net_socket* socket, void* buffer, size_t size, size_t* size_recieved);
-bool CGL_net_socket_shutdown_send(CGL_net_socket* socket);
-bool CGL_net_socket_shutdown_recv(CGL_net_socket* socket);
-
-int CGL_net_http_request(const char* method, const char* host, const char* path, void* response_buffer, size_t* size, const char* accept, const char* user_agent, const char* body);
-int CGL_net_http_get(const char* host, const char* path, void* buffer, size_t* size, const char* accept, const char* user_agent);
+
+
+/**
+ * It Connects a socket to a target address.
+ *
+ * @param socket  The socket which we want to connect
+ * @param target  The target address on which we want to connect
+ *
+ * @return IT will return true if connection was successful or else it will give out false
+ */
+bool CGL_net_socket_connect(CGL_net_socket* socket, CGL_net_addrinfo* target)
+{
```

Comment:
The comment format is more or less correct.
Problems are no brief for the function is given.
Why is a function body added here? its wrong
Check this part of code has already been documented ( https://github.com/Jaysmito101/cgl/commit/00b84d613f0cea61181526f97f7af2cffbc134bf)